plugin/advisor/pg/advisor_naming_unique_key_convention.go

```diff
@@ -0,0 +1,234 @@
+				Code:    advisor.NamingUKConventionMismatch,
+				Title:   checker.title,
+				Content: fmt.Sprintf(`Unique key in table "%s" mismatches the naming convention, expect %q but found "%s"`, indexData.tableName, regex, indexData.indexName),
+			})
+		}
+		if checker.maxLength > 0 && len(indexData.indexName) > checker.maxLength {
+			checker.adviceList = append(checker.adviceList, advisor.Advice{
+				Status:  checker.level,
+				Code:    advisor.NamingUKConventionMismatch,
+				Title:   checker.title,
+				Content: fmt.Sprintf(`Unique key "%s" in table "%s" mismatches the naming convention, its length should be within %d characters`, indexData.indexName, indexData.tableName, checker.maxLength),
+			})
+		}
+	}
+
+	return checker
+}
+
+// getMetaDataList returns the list of unique key with metadata.
+func (checker *namingUKConventionChecker) getMetaDataList(in ast.Node) []*indexMetaData {
+	var res []*indexMetaData
+	switch node := in.(type) {
+	case *ast.CreateTableStmt:
+		for _, constraint := range node.ConstraintList {
+			if metadata := checker.getUniqueKeyMetadata(constraint, node.Name.Name); metadata != nil {
+				res = append(res, metadata)
+			}
+		}
+		for _, column := range node.ColumnList {
+			for _, constraint := range column.ConstraintList {
+				if metadata := checker.getUniqueKeyMetadata(constraint, node.Name.Name); metadata != nil {
+					res = append(res, metadata)
+				}
+			}
+		}
+	case *ast.AddConstraintStmt:
+		constraint := node.Constraint
+		if metadata := checker.getUniqueKeyMetadata(constraint, node.Table.Name); metadata != nil {
+			res = append(res, metadata)
+		}
+	case *ast.AddColumnListStmt:
+		for _, column := range node.ColumnList {
+			for _, constraint := range column.ConstraintList {
+				if metadata := checker.getUniqueKeyMetadata(constraint, node.Table.Name); metadata != nil {
+					res = append(res, metadata)
+				}
+			}
+		}
+	case *ast.CreateIndexStmt:
+		if node.Index.Unique {
+			var columnList []string
+			for _, key := range node.Index.KeyList {
+				columnList = append(columnList, key.Key)
+			}
+			metaData := map[string]string{
+				advisor.ColumnListTemplateToken: strings.Join(columnList, "_"),
+				advisor.TableNameTemplateToken:  node.Index.Table.Name,
+			}
+			res = append(res, &indexMetaData{
+				indexName: node.Index.Name,
+				tableName: node.Index.Table.Name,
+				metaData:  metaData,
+			})
+		}
+	case *ast.RenameConstraintStmt:
+		if index := checker.findIndex(context.Background(), node.Table.Name, node.ConstraintName); index != nil && index.Unique {
+			metaData := map[string]string{
+				advisor.ColumnListTemplateToken: strings.Join(index.ColumnExpressions, "_"),
+				advisor.TableNameTemplateToken:  node.Table.Name,
+			}
+			res = append(res, &indexMetaData{
+				indexName: node.NewName,
+				tableName: node.Table.Name,
+				metaData:  metaData,
+			})
+		}
+	case *ast.RenameIndexStmt:
+		// "ALTER INDEX name RENAME TO new_name" doesn't take a table name
```

Comment:
In PostgreSQL, index name is unique in a schema, not a table. So I think it's OK here.
![image](https://user-images.githubusercontent.com/20775801/180587860-f5413e05-0364-4ffa-8a53-2cf4024e87b9.png)
details in https://stackoverflow.com/questions/27306539/at-what-level-do-postgres-index-names-need-to-be-unique

Reply:
We can write it like this for now and fix it here after I finish the refactoring in https://github.com/bytebase/bytebase/pull/1992. Please add a TODO comment to me here.

Reply:
Added.